src/server_manager/web_app/app.ts

```diff
@@ -373,18 +372,17 @@ export class App {
   }
 
   private connectToDigitalOcean() {
+    const oauthUi = this.appRoot.getAndShowDigitalOceanOauthFlow();
```

Comment:
We can pass a `onCancel` callback here too. You'll need to move this to after `runDigitalOceanOauth`